### ISSUE TITLE
Review the role of GCIs with complex LHS in base extraction

### ISSUE THREAD

**Issue by matentzn** [INCATools/ontology-development-kit#836]
We have recently finally defined a recommendation for shaping base files: https://github.com/OBOAcademy/obook/pull/387

The one thing that is still not addressed is what to do about GCIs with complex RHS, which are used plentiful in Uberon and PATO. 

Currently, these are _not_ included in the base files. We need to review that decision to be sure that this is the right way to go!

We could potentially use some kind of tagging approach to force certain axioms into the base, even if they are not covered by the general process.

cc @dosumis @cmungall @balhoff

**Comment by matentzn** [comment]
(@balhoff this is important for you because of the role these GCIs may play in Ubergraph)

**Comment by gouttegd** [comment]
Poor GCIs. Nobody ever seems to care about them…

**Comment by balhoff** [comment]
I do think they should be included. We have run into similar problems with things like SWRL rules. It seems to me that the base file should include all the asserted axioms in the ontology's source. My understanding is that the challenge comes from using a subtractive process to create the base file from a merged product. Perhaps there should be another step merging sources back in after reasoning and filtering?

**Comment by matentzn** [comment]
> It seems to me that the base file should include all the asserted axioms in the ontology's source.

This is not a formal definition.. The goal for base file spec is to provide a formalisation of what they are so you can compute them (modularisation) even if an ontology does not provide a base. So we still need a formalisation that allows us to select these axioms for a reductive approach, eg.:

1. Axiom is tagged with `rdfs:isDefinedBy <..mondo.owl>` - precise but requires curation (I prefer this atm)
2. Axiom LHS contains at least one base entity - automatic, but possibly redundant (two independent ontologies like RO and Uberon would think it belongs in its base.

**Comment by balhoff** [comment]
I agree we need ways to create a base file when one isn't provided, but I still think that's a workaround for the original goal of base, which to me is exactly my definition. I don't see what's not formal about that definition—it's more about responsibility for axioms rather than signatures.

**Comment by matentzn** [comment]
For example because it defeats one major purpose: you could decide you WANT to axiom inject in edit - that axiom should still not appear in the base.
